std/haxe/macro/Expr.hx

```diff
@@ -169,154 +213,603 @@ enum Unop {
+	**/
 	ECall( e : Expr, params : Array<Expr> );
+
+	/**
+		A constructor call `new t(params)`.
+	**/
 	ENew( t : TypePath, params : Array<Expr> );
+
+	/**
+		An unary operator `op` on `e`:
+		
+		* e++ (op = OpIncrement, postFix = true)
+		* e-- (op = OpDecrement, postFix = true)
+		* ++e (op = OpIncrement, postFix = false)
+		* --e (op = OpDecrement, postFix = false)
+		* -e (op = OpNeg, postFix = false)
+		* !e (op = OpNot, postFix = false)
+		* ~e (op = OpNegBits, postFix = false)
+	**/
 	EUnop( op : Unop, postFix : Bool, e : Expr );
+
+	/**
+		A variable declaration `var v` or `var v = expr`.
+	**/
 	EVars( vars : Array<Var> );
+
+	/**
+		A function declaration.
+	**/
 	EFunction( name : Null<String>, f : Function );
+
+	/**
+		A block declaration `{exprs}`.
+	**/
 	EBlock( exprs : Array<Expr> );
+
+	/**
+		A `for` expression.
+	**/
 	EFor( it : Expr, expr : Expr );
+
+	/**
+		A `(e1 in e2)` expression.
+	**/
 	EIn( e1 : Expr, e2 : Expr );
+
+	/**
+		An `if(econd) eif` or `if(econd) eif else eelse` expression.
+	**/
 	EIf( econd : Expr, eif : Expr, eelse : Null<Expr> );
+
+	/**
+		Represents a `while` expression.
+		When `normalWhile` is `true` it is `while (...)`.
+		When `normalWhile` is `false` it is `do {...} while (...)`.
+	**/
 	EWhile( econd : Expr, e : Expr, normalWhile : Bool );
+
+	/**
+		Represents a `switch` expression with related cases and an optional.
+		`default` case if edef != null.
+	**/
 	ESwitch( e : Expr, cases : Array<Case>, edef : Null<Null<Expr>> );
+
+	/**
+		Represents a `try`-expression with related catches.
+	**/
 	ETry( e : Expr, catches : Array<Catch> );
+
+	/**
+		A `return` or `return e` expression.
+	**/
 	EReturn( ?e : Null<Expr> );
+
+	/**
+		A `break` expression.
+	**/
 	EBreak;
+
+	/**
+		A `continue` expression.
+	**/
 	EContinue;
+
+	/**
+		A `untyped e` expression.
+	**/
 	EUntyped( e : Expr );
+
+	/**
+		A `throw e` expression.
+	**/
 	EThrow( e : Expr );
+
+	/**
+		A `cast e` or `cast (e, m)` expression.
+	**/
 	ECast( e : Expr, t : Null<ComplexType> );
+
+	/**
+		Internally used to provide completion.
+	**/
 	EDisplay( e : Expr, isCall : Bool );
+
+	/**
+		Internally used to provide completion.
+	**/
 	EDisplayNew( t : TypePath );
+
+	/**
+		A `(econd) ? eif : eelse` expression.
+	**/
 	ETernary( econd : Expr, eif : Expr, eelse : Expr );
+
+	/**
+		A `(e:t)` expression.
+	**/
 	ECheckType( e : Expr, t : ComplexType );
+
+	/**
+		A `@m e` expression.
+	**/
 	EMeta( s : MetadataEntry, e : Expr );
 }
 
+/**
+	Represents a complex type for the AST.
+**/
 enum ComplexType {
+	/**
+		Represents the type path.
+	**/
 	TPath( p : TypePath );
+
+	/**
+		Represents a function type.
+		@see http://haxe.org/manual/types-function.html
+	**/
 	TFunction( args : Array<ComplexType>, ret : ComplexType );
+
+	/**
+		Represents an anonymous structure type.
+		@see http://haxe.org/manual/types-anonymous-structure.html
+	**/
 	TAnonymous( fields : Array<Field> );
+
+	/**
+		Represents the parent type.
```

Comment:
While the thought of types having parents is entertaining, this is actually about parentheses, e.g. in `(Int -> Void) -> Void` or something.